core/math/math_funcs.cpp

```diff
@@ -64,37 +64,37 @@ uint32_t Math::rand() {
 	return rand_from_seed(&default_seed);
 }
 
-double Math::randf() {
+real_t Math::randf() {
 
```

Comment:
would leave these as a double, remember GDScript scalars are doubles even if real_t is float.